app/src/main/java/vn/edu/usth/minigh/AuthActivity.kt

```diff
@@ -1,21 +1,176 @@
+            }
+        }
+    }
+
+
+    fun requestForAccessToken(code: String) {
+        val grantType = "authorization_code"
+
+        val postParams =
+                "grant_type=" + grantType + "&code=" + code + "&redirect_uri=" + GithubConstants.REDIRECT_URI + "&client_id=" + GithubConstants.CLIENT_ID + "&client_secret=" + GithubConstants.CLIENT_SECRET
+        GlobalScope.launch(Dispatchers.Default) {
+            val url = URL(GithubConstants.TOKEN_URL)
+            val httpsURLConnection =
+                    withContext(Dispatchers.IO) { url.openConnection() as HttpsURLConnection }
+            httpsURLConnection.requestMethod = "POST"
+            httpsURLConnection.setRequestProperty(
+                    "Accept",
+                    "application/json"
+            );
+            httpsURLConnection.doInput = true
+            httpsURLConnection.doOutput = true
+            val outputStreamWriter = OutputStreamWriter(httpsURLConnection.outputStream)
+            withContext(Dispatchers.IO) {
+                outputStreamWriter.write(postParams)
+                outputStreamWriter.flush()
+            }
+            val response = httpsURLConnection.inputStream.bufferedReader()
+                    .use { it.readText() }  // defaults to UTF-8
+            withContext(Dispatchers.Main) {
+                val jsonObject = JSONTokener(response).nextValue() as JSONObject
+
+                val accessToken = jsonObject.getString("access_token") //The access token
+
+                // Get user's id, first name, last name, profile pic url
+                fetchGithubUserProfile(accessToken)
+            }
+        }
+    }
+
+    fun fetchGithubUserProfile(token: String) {
+        GlobalScope.launch(Dispatchers.Default) {
+            val tokenURLFull =
+                    "https://api.github.com/user"
+
+            val url = URL(tokenURLFull)
+            val httpsURLConnection =
+                    withContext(Dispatchers.IO) { url.openConnection() as HttpsURLConnection }
+            httpsURLConnection.requestMethod = "GET"
+            httpsURLConnection.setRequestProperty("Authorization", "Bearer $token")
+            httpsURLConnection.doInput = true
+            httpsURLConnection.doOutput = false
+            val response = httpsURLConnection.inputStream.bufferedReader()
+                    .use { it.readText() }  // defaults to UTF-8
+            val jsonObject = JSONTokener(response).nextValue() as JSONObject
+            Log.i("GitHub Access Token: ", token)
+            accessToken = token
+
+            // GitHub Id
+            val githubId = jsonObject.getInt("id")
+            Log.i("GitHub Id: ", githubId.toString())
+            id = githubId.toString()
+
+            // GitHub Display Name
+            val githubDisplayName = jsonObject.getString("login")
+            Log.i("GitHub Display Name: ", githubDisplayName)
+            displayName = githubDisplayName
+        }
     }
```

Comment:
oi @Huy-Ngo ,@PhuongPhg and @s0ys4uc3, you can take this as example for getting stuff